gluster/block/glusterblock-provisioner.go

```diff
@@ -142,25 +146,35 @@ func (p *glusterBlockProvisioner) Provision(options controller.VolumeOptions) (*
 
 	glog.V(4).Infof("glusterfs: creating volume with configuration %+v", p.provConfig)
 
-	// Create unique PVC identity.
-	blockVolIdentity := fmt.Sprintf("kubernetes-dynamic-pvc-%s", uuid.NewUUID())
-
-	//TODO: blockVol := volPrefix + string(uuid.NewUUID())
-	vol, err := p.createVolume(blockVolIdentity)
+	blockVol := volPrefix + string(uuid.NewUUID())
+	vol, err := p.createVolume(blockVol)
 	if err != nil {
 		return nil, err
 	}
 
-	// TODO:
-	vol.TargetPortal = vol.Portals[0]
-	nameSpace := options.PVC.Namespace
+	if len(vol.Portals) == 1 && vol.Portals[0] != "" {
+		vol.TargetPortal = vol.Portals[0]
+		vol.Portals = nil
+	} else {
+		for _, v := range vol.Portals {
+			if v != "" && vol.TargetPortal == "" {
+				vol.TargetPortal = v
+				continue
+			} else {
+				vol.Portals = append(vol.Portals, v)
```

Comment:
what is the point of appending to itself? What happens if there is no vol.TargetPortal set?